### PR TITLE
HPCC-26449 User Persona in top nav

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <jeremy.clements@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/MyAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    employeeID: string;
+    firstName: string;
+    lastName: string;
+    passwordDaysRemaining: number;
+    passwordExpiration: string;
+    passwordExpirationWarningDays: number;
+    passwordIsExpired: boolean;
+    passwordNeverExpires: boolean;
+    username: string;
+};
```
Comment (GordonSmith) [inline esp/src/src-react/components/MyAccount.tsx:23]: Replace with version in "@hpcc-js/comms" v1.57.0

**Review by GordonSmith — CHANGES_REQUESTED** [review]
Looks good

File: esp/src/src-react/components/MyAccount.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                    setOldPassword(value);
+                    break;
+                case "newPassword1":
+                    setNewPassword1(value);
+                    break;
+                case "newPassword2":
+                    setNewPassword2(value);
+                    break;
+                default:
+                    console.log(id, value);
```
Comment (GordonSmith) [inline esp/src/src-react/components/MyAccount.tsx:87]: Remove logging code (or use `logger.debug`

File: esp/src/src-react/components/Title.tsx [hunk 3]
```diff
@@ -62,18 +76,46 @@ export const DevTitle: React.FunctionComponent<DevTitleProps> = ({
...
+                        fetch("esp/logout", {
+                            method: "post"
+                        }).then(data => {
+                            if (data) {
+                                cookie("ECLWatchUser", "", { expires: -1 });
+                                cookie("ESPSessionID" + location.port + " = '' ", "", { expires: -1 });
+                                cookie("Status", "", { expires: -1 });
+                                cookie("User", "", { expires: -1 });
+                                window.location.reload();
+                            }
```
Comment (GordonSmith) [inline esp/src/src-react/components/Title.tsx:97]: @jeclrsg can you open a Jira to add a cookie warning / approval?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/MyAccount.tsx (hunk 2 above)
Comment (jeclrsg) [reply esp/src/src-react/components/MyAccount.tsx]: Changed to logger.debug(...)

File: esp/src/src-react/components/Title.tsx (hunk 3 above)
Comment (jeclrsg) [reply esp/src/src-react/components/Title.tsx:97]: Issue opened (HPCC-26627)

**Review by GordonSmith — APPROVED** [review]
Looks good

**Comment by richardkchapman** [comment]
Needs rebasing

**Comment by GordonSmith** [comment]
@richardkchapman I have resolved the conflict

**Pull request merged**